Impossible to build the dockerfile

Hi,
First of all, thank you for this project.

I saw your e-mail on the SKS mailing list and wanted to give it a try.

However I can't manage to build the dockerfile:
```
$ docker build -t hockeypuck .
Sending build context to Docker daemon  160.5MB
Step 1/13 : FROM golang:1.12 AS builder
 ---> ffcaee6f7d8b
Step 2/13 : COPY src /hockeypuck/src
 ---> f4192f535c61
Step 3/13 : ENV GOPATH=/hockeypuck
 ---> Running in cda62241fe8d
Removing intermediate container cda62241fe8d
 ---> f7c158733696
Step 4/13 : WORKDIR /hockeypuck/src
 ---> Running in 8a7d5778a86d
Removing intermediate container 8a7d5778a86d
 ---> cac2af7982ff
Step 5/13 : RUN go vet ./...
 ---> Running in f938dcb467bc
# hockeypuck/conflux [hockeypuck/conflux.test]
hockeypuck/conflux/decode_test.go:346:11: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
# hockeypuck/hkp/storage
hockeypuck/hkp/storage/storage.go:33:9: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
# hockeypuck/conflux/recon
hockeypuck/conflux/recon/gossip.go:62:9: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:71:9: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:160:7: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:241:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:248:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:288:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/peer.go:340:7: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
# hockeypuck/conflux/recon [hockeypuck/conflux/recon.test]
hockeypuck/conflux/recon/gossip.go:62:9: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:71:9: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:160:7: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:241:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:248:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/gossip.go:288:5: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
hockeypuck/conflux/recon/peer.go:340:7: undefined: "hockeypuck/vendor/github.com/pkg/errors".Is
The command '/bin/sh -c go vet ./...' returned a non-zero code: 2
```

Sadly I'm not yet proficient in go in order to be able to fix this...

Cheers,
Frank

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,5 @@
+/bin
+/pkg
+/contrib/pyinfra
+/.git
+/**/secrets

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,14 +1,25 @@
-FROM golang:1.12 AS builder
-COPY src /hockeypuck/src
+FROM golang:1.15 AS toolchain
+
+FROM ubuntu:18.04 as builder
+RUN adduser builder --system --disabled-login
+COPY --from=toolchain /usr/local/go /usr/local/go
+ENV PATH="/usr/local/go/bin:/usr/lib/postgresql/10/bin:${PATH}"
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -qq && apt-get -y install build-essential postgresql-10 postgresql-server-dev-10 ca-certificates --no-install-recommends
+COPY --chown=builder:root . /hockeypuck
 ENV GOPATH=/hockeypuck
-WORKDIR /hockeypuck/src
-RUN go vet ./...
-RUN go install ./hockeypuck/server/cmd/...
+USER builder
+WORKDIR /hockeypuck
+RUN make lint test test-postgresql
+RUN go install hockeypuck/server/cmd/...
+ENTRYPOINT ["/bin/bash"]
+
 
 FROM ubuntu:18.04
+RUN apt-get update -qq && apt-get dist-upgrade -y && apt-get -y install ca-certificates --no-install-recommends
 RUN mkdir -p /hockeypuck/bin /hockeypuck/lib /hockeypuck/etc /hockeypuck/data
 COPY --from=builder /hockeypuck/bin /hockeypuck/bin
-COPY contrib/templates /hockeypuck/lib/templates
-COPY contrib/webroot /hockeypuck/lib/www
+COPY --from=builder /hockeypuck/contrib/templates /hockeypuck/lib/templates
+COPY --from=builder /hockeypuck/contrib/webroot /hockeypuck/lib/www
 VOLUME /hockeypuck/etc /hockeypuck/data
 ENTRYPOINT ["/hockeypuck/bin/hockeypuck", "-config", "/hockeypuck/etc/hockeypuck.conf"]

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -15,7 +15,7 @@ services:
     - hkp_data:/hockeypuck/data
 
   postgres:
-    image: postgres:11
+    image: postgres
     restart: always
     environment:
     - POSTGRES_USER=docker

